### PR TITLE
Feature/group selection screen

### PR DESCRIPTION
**Pull request by krida2000**

### CODE REVIEW

File: src/features/group_selection_screen/lib/models/group.dart [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import 'subgroup.dart';
+
+class Group {
```
Comment (extreamer-lynx) [inline src/features/group_selection_screen/lib/models/group.dart:3]: Make test

File: src/features/group_selection_screen/lib/models/subgroup.dart [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Subgroup {
+  Subgroup({
+    this.id,
```
Comment (extreamer-lynx) [inline src/features/group_selection_screen/lib/models/subgroup.dart:3]: Make test

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                    height: 20,
+                  ),
+                  DropdownButton<int>(
+                    hint: Text('Course'),
+                    isExpanded: true,
+                    value: course,
+                    onChanged: (int? newValue) {
+                      setState(() {
+                        groups = [
+                          Group(name: 'group_1_' + newValue!.toString()),
```
Comment (extreamer-lynx) [inline src/features/group_selection_screen/lib/screens/group_selection_screen.dart:67]: Get data ftom firestore

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                  SizedBox(
+                    height: 15,
+                  ),
+                  group != null &&
+                          group!.subgroups != null &&
+                          group!.subgroups!.length > 0
+                      ? Column(
+                          children: [
+                            DropdownButton<Subgroup>(
+                              hint: Text('Subgroup'),
```
Comment (extreamer-lynx) [inline src/features/group_selection_screen/lib/screens/group_selection_screen.dart:128]: Move to constant class

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                          value: isMyGroup,
+                          onChanged: (value) {
+                            setState(() {
+                              isMyGroup = value!;
+                            });
+                          }),
+                      GestureDetector(
+                        child: Padding(
+                          padding: const EdgeInsets.all(8.0),
+                          child: Text('It`s my group'),
```
Comment (extreamer-lynx) [inline src/features/group_selection_screen/lib/screens/group_selection_screen.dart:166]: Move to constant class

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+                        value: value,
+                        child: Text(value.toString()),
+                      );
+                    }).toList(),
+                  ),
+                  SizedBox(
+                    height: 15,
+                  ),
+                  DropdownButton<Group>(
+                    hint: Text('Group'),
```
Comment (extreamer-lynx) [inline src/features/group_selection_screen/lib/screens/group_selection_screen.dart:101]: Move to constant class

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+                  Text(
+                    'Chose course and group',
+                    textAlign: TextAlign.center,
+                    textScaleFactor: 3,
+                  ),
+                  SizedBox(
+                    height: 20,
+                  ),
+                  DropdownButton<int>(
+                    hint: Text('Course'),
```
Comment (extreamer-lynx) [inline src/features/group_selection_screen/lib/screens/group_selection_screen.dart:61]: Move to constant class

File: src/features/group_selection_screen/android/build.gradle [hunk 8]
```diff
@@ -0,0 +1,40 @@
+group 'com.example.group_selection_screen'
```
[the remark this answers is not in the document]
Comment (krida2000) [reply src/features/group_selection_screen/android/build.gradle:1]: +

File: src/features/group_selection_screen/lib/models/group.dart (hunk 1 above)
Comment (krida2000) [reply src/features/group_selection_screen/lib/models/group.dart:3]: +

File: src/features/group_selection_screen/lib/models/subgroup.dart (hunk 2 above)
Comment (krida2000) [reply src/features/group_selection_screen/lib/models/subgroup.dart:3]: +

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart (hunk 3 above)
Comment (krida2000) [reply src/features/group_selection_screen/lib/screens/group_selection_screen.dart:67]: +

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart (hunk 4 above)
Comment (krida2000) [reply src/features/group_selection_screen/lib/screens/group_selection_screen.dart:128]: +

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart (hunk 5 above)
Comment (krida2000) [reply src/features/group_selection_screen/lib/screens/group_selection_screen.dart:170]: +

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart (hunk 6 above)
Comment (krida2000) [reply src/features/group_selection_screen/lib/screens/group_selection_screen.dart:101]: +

File: src/features/group_selection_screen/lib/screens/group_selection_screen.dart (hunk 7 above)
Comment (krida2000) [reply src/features/group_selection_screen/lib/screens/group_selection_screen.dart:61]: +

File: src/features/group_selection/README.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# group_selection
+
+A new Flutter package project.
+
+## Getting Started
+
+This project is a starting point for a Dart
```
[the remark this answers is not in the document]
Comment (krida2000) [reply src/features/group_selection/README.md:7]: +

File: src/features/group_selection/LICENSE [hunk 10]
```diff
@@ -0,0 +1 @@
+TODO: Add your license here.
```
[the remark this answers is not in the document]
Comment (krida2000) [reply src/features/group_selection/LICENSE:1]: +

File: src/assets/translations/uk.json [hunk 11]
```diff
@@ -1,3 +1,10 @@
 {
-    "Zhytomyr Polytechnic": "Житомирська Політехніка"
+    "Zhytomyr Polytechnic": "Житомирська Політехніка",
+    "Schedule": "Розклад",
+    "Chose course and group": "Оберіть курс та групу",
+    "Course": "Курс",
+    "Group": "Група",
```
[the remark this answers is not in the document]
Comment (krida2000) [reply src/assets/translations/uk.json:6]: +

File: src/features/group_selection/lib/src/abstract/firebase_data_getter.dart [hunk 12]
```diff
@@ -0,0 +1,35 @@
+import '../models/models.dart';
+
+abstract class FirebaseDataGetter {
+  Future<List<Group>> getGroups(int course);
+}
+
+class FirebaseDataGetterMock implements FirebaseDataGetter {
+  Future<List<Group>> getGroups(int course) async {
+    await Future.delayed(Duration(seconds: 1), () {});
```
[the remark this answers is not in the document]
Comment (krida2000) [reply src/features/group_selection/lib/src/abstract/firebase_data_getter.dart:9]: +

**Pull request merged**